packages/custom-elements/src/Patch/Node.ts

```diff
@@ -0,0 +1,227 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt The complete set of authors may be found
+ * at http://polymer.github.io/AUTHORS.txt The complete set of contributors may
+ * be found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by
+ * Google as part of the polymer project is also subject to an additional IP
+ * rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import CustomElementInternals from '../CustomElementInternals.js';
+import * as Utilities from '../Utilities.js';
+
+import * as Native from './Native.js';
+
+export default function(internals: CustomElementInternals) {
+  // `Node#nodeValue` is implemented on `Attr`.
+  // `Node#textContent` is implemented on `Attr`, `Element`.
+
+  Node.prototype.insertBefore = function(this: Node, node, refNode) {
+    if (node instanceof DocumentFragment) {
+      const insertedNodes = Utilities.childrenFromFragment(node);
+      const nativeResult = Native.Node_insertBefore.call(this, node, refNode);
+
+      // DocumentFragments can't be connected, so `disconnectTree` will never
+      // need to be called on a DocumentFragment's children after inserting it.
+
+      if (Utilities.isConnected(this)) {
+        for (let i = 0; i < insertedNodes.length; i++) {
+          internals.connectTree(insertedNodes[i]);
+        }
+      }
+
+      return nativeResult;
+    }
+
+    const nodeWasConnectedElement =
+        node instanceof Element && Utilities.isConnected(node);
+    const nativeResult = Native.Node_insertBefore.call(this, node, refNode);
+
+    if (nodeWasConnectedElement) {
+      internals.disconnectTree(node);
+    }
+
+    if (Utilities.isConnected(this)) {
+      internals.connectTree(node);
+    }
+
+    return nativeResult;
+  } as typeof Node.prototype.insertBefore;
```

Comment:
It's because calling `nativeVersion.call(...)` doesn't properly pass the generics through. Added casts on the calls to native versions and removed them from the patch assignment statements.